internal/core/common_config/main.go

```diff
@@ -76,22 +76,22 @@ func Main(ctx context.Context, cancel context.CancelFunc) {
 	var accessToken string
 	var getAccessToken types.GetAccessTokenCallback
 
-	if secretProvider != nil {
-		getAccessToken = func() (string, error) {
-			accessToken, err = secretProvider.GetAccessToken("consul", common.CoreCommonConfigServiceKey)
-			if err != nil {
-				return "", fmt.Errorf("failed to get Configuration Provider access token: %s", err.Error())
-			}
-			return accessToken, err
-		}
-		_, err = getAccessToken()
+	// need to use in-line function to set the callback type for getAccessToken used in CreateProviderClient to allow
+	// access to the config provider in secure mode
+	getAccessToken = func() (string, error) {
+		accessToken, err = secretProvider.GetAccessToken("consul", common.CoreCommonConfigServiceKey)
 		if err != nil {
-			lc.Errorf("failed to get Access Token for config provider: %s", err.Error())
-			os.Exit(1)
+			return "", fmt.Errorf("failed to get Configuration Provider access token: %s", err.Error())
 		}
+		return accessToken, err
+	}
+	_, err = getAccessToken()
+	if err != nil {
+		lc.Errorf("failed to get Access Token for config provider: %s", err.Error())
+		os.Exit(1)
```

Comment:
I think this can be removed also since CreateProviderClient  is calling it.

Reply:
ok. didn't know if we needed this error handling or not.